File: conversions/typescript/Collective.ts
import { BaseElement } from './base_elements';

export interface Collective extends BaseElement {
  // Formation
  composition?: string | null;
  count?: number | null;
  formationDate?: number | null;
  operatorId?: string | null;
  equipmentIds?: string[] | null;
  // Agency
  activity?: string | null;
  temperance?: string | null;
  skillsIds?: string[] | null;
  ritualsIds?: string[] | null;
  // World
  speciesIds?: string[] | null;
  charactersIds?: string[] | null;
  creaturesIds?: string[] | null;
  phenomenaIds?: string[] | null;
}

File: languages/csharp_unity/Collective.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Collective : BaseElement
{
    // Formation
    public string Composition;
    public int? Count;
    public int? FormationDate;
    public string OperatorId;
    public List<string> EquipmentIds; = new List<string>();
    // Dynamics
    public string Activity;
    public string Disposition;
    public string State;
    public List<string> AbilitiesIds; = new List<string>();
    public List<string> SymbolismIds; = new List<string>();
    // World
    public List<string> SpeciesIds; = new List<string>();
    public List<string> CharactersIds; = new List<string>();
    public List<string> CreaturesIds; = new List<string>();
    public List<string> PhenomenaIds; = new List<string>();
}
